### PR TITLE
Cleanup Unnecessary Nodepool/Nodeclaim util functions and inline them

### LINKED ISSUE #893: Remove `utils/nodeclaim` and `utils/nodepool` functions
### Description

**What problem are you trying to solve?**

There are some leftover helper functions in the repo that we should clean-up from the v1beta1 migration. Most notably, anything that has to do with metrics or `kubeClient` calls in these packages can just be directly moved up into the calling code, removing any need for the util abstraction.

**How important is this feature to you?**

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by sadath-12**
Fixes #893  

**Description**

Cleanup of functions ```utils/nodeclaim``` and ```utils/nodepool``` that can be instead used as inline on the main code.

If you were previously using any of the `nodepoolutil` or `nodeclaimutil` package functions, you can migrate to calling the `client.Client` directly. These functions no longer add value since the v1beta1 migration since these functions were meant to abstract away model -> view conversion during the migration. With v1beta1, the model and the view are now the same (the v1beta1 apis).

**How was this change tested?**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/utils/nodeclaim/suite_test.go [hunk 1]
```diff
@@ -161,7 +163,7 @@ var _ = Describe("NodeClaimUtils", func() {
 
 		providerID := test.RandomProviderID()
 		nodeClaim.Status.ProviderID = providerID
-		Expect(nodeclaimutil.UpdateStatus(ctx, env.Client, nodeClaim)).To(Succeed())
+		Expect(env.Client.Status().Update(ctx, nodeClaim)).To(Succeed())
```
Comment (jonathan-innis) [inline pkg/utils/nodeclaim/suite_test.go:166]: If we're dropping these functions, then these tests aren't really testing anything anymore so I think that we can drop them

File: pkg/utils/nodepool/nodepool.go [hunk 2]
```diff
@@ -40,15 +40,3 @@ func List(ctx context.Context, c client.Client, opts ...client.ListOption) (*v1b
 	}
 	return nodePoolList, nil
 }
-
```
Comment (jonathan-innis) [inline pkg/utils/nodepool/nodepool.go:43]: Ideally, I think we can drop this file entirely if we are removing the utils here

File: pkg/utils/nodepool/suite_test.go [hunk 3]
```diff
@@ -33,11 +33,12 @@ import (
 	"sigs.k8s.io/karpenter/pkg/operator/scheme"
 	"sigs.k8s.io/karpenter/pkg/test"
 	. "sigs.k8s.io/karpenter/pkg/test/expectations"
-	nodepoolutil "sigs.k8s.io/karpenter/pkg/utils/nodepool"
 )
 
-var ctx context.Context
-var env *test.Environment
+var (
```
Comment (jonathan-innis) [inline pkg/utils/nodepool/suite_test.go:38]: If we drop the NodePool utils file, we should also be able to drop the testing here

File: pkg/controllers/nodeclaim/lifecycle/controller.go [hunk 4]
```diff
@@ -40,9 +40,9 @@ import (
 
 	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
 	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	nodeclaimutil "sigs.k8s.io/karpenter/pkg/utils/nodeclaim"
```
Comment (jonathan-innis) [inline pkg/controllers/nodeclaim/lifecycle/controller.go:43]: Do we even need the NodeClaimUtil anymore if we are dropping the other helpers?

File: pkg/controllers/nodeclaim/disruption/expiration.go [hunk 5]
```diff
@@ -68,7 +68,10 @@ func (e *Expiration) Reconcile(ctx context.Context, nodePool *v1beta1.NodePool,
 	})
 	if !hasExpiredCondition {
 		logging.FromContext(ctx).Debugf("marking expired")
-		nodeclaimutil.DisruptedCounter(nodeClaim, metrics.ExpirationReason).Inc()
+		metrics.NodeClaimsDisruptedCounter.With(prometheus.Labels{
+			metrics.TypeLabel:     metrics.ExpirationReason,
+			metrics.NodePoolLabel: nodeClaim.Labels[v1beta1.NodePoolLabelKey],
+		})
```
Comment (jonathan-innis) [inline pkg/controllers/nodeclaim/disruption/expiration.go:74]: Missed the `.Inc()` call here

**Comment by jonathan-innis** [comment]
@sadath-12 Looks like you need to run `make presubmit` once; without it, you're getting CI failures.

**Comment by sadath-12** [comment]
Done @jonathan-innis

File: pkg/controllers/provisioning/provisioner.go [hunk 6]
```diff
@@ -207,9 +207,10 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNod
 	instanceTypes := map[string][]*cloudprovider.InstanceType{}
 	domains := map[string]sets.Set[string]{}
 
-	nodePoolList, err := nodepoolutil.List(ctx, p.kubeClient)
+	nodePoolList := &v1beta1.NodePoolList{}
+	err := p.kubeClient.List(ctx, nodePoolList)
```
Comment (jonathan-innis) [inline pkg/controllers/provisioning/provisioner.go:211]: nit: could collapse L211 and L212 down into a single line with 

```
if err := p.kubeClient.List(ctx, nodePoolList); err != nil {
```

File: pkg/controllers/nodeclaim/garbagecollection/controller.go [hunk 7]
```diff
@@ -55,7 +56,8 @@ func (c *Controller) Name() string {
 }
 
 func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
-	nodeClaimList, err := nodeclaimutil.List(ctx, c.kubeClient)
+	nodeClaimList := &v1beta1.NodeClaimList{}
+	err := c.kubeClient.List(ctx, nodeClaimList)
```
Comment (jonathan-innis) [inline pkg/controllers/nodeclaim/garbagecollection/controller.go:60]: nit: Could combine L60 and L61 into a single line

File: pkg/controllers/nodeclaim/disruption/controller.go [hunk 8]
```diff
@@ -110,14 +112,30 @@ func (c *Controller) Name() string {
 	return "nodeclaim.disruption"
 }
 
+// NodePoolEventHandler is a watcher on v1beta1.NodeClaim that maps Provisioner to NodeClaims based
+// on the v1beta1.NodePoolLabelKey and enqueues reconcile.Requests for the NodeClaim
+func NodePoolEventHandler(c client.Client) handler.EventHandler {
```
Comment (jonathan-innis) [inline pkg/controllers/nodeclaim/disruption/controller.go:117]: Not sure that we need to move this into here in this PR. Let's keep the event handlers in one place in the `nodeclaimutil` package for now

File: pkg/controllers/disruption/orchestration/queue.go [hunk 9]
```diff
@@ -222,7 +226,8 @@ func (q *Queue) waitOrTerminate(ctx context.Context, cmd *Command) error {
 			continue
 		}
 		// Get the nodeclaim
-		nodeClaim, err := nodeclaim.Get(ctx, q.kubeClient, cmd.Replacements[i].name)
+		nodeClaim := &v1beta1.NodeClaim{}
+		err := q.kubeClient.Get(ctx, types.NamespacedName{Name: cmd.Replacements[i].name}, nodeClaim)
```
Comment (jonathan-innis) [inline pkg/controllers/disruption/orchestration/queue.go:230]: nit: Same comment here. I think we can combine L230 and L231

**Comment by jonathan-innis** [comment]
@sadath-12 The other thing here is that we need to keep in mind that these utils functions are being used by the CloudProvider's as well. We should try and open PRs against `Azure/karpenter` and `aws/karpenter-provider-aws` that port over this change after we merge in this cleanup

**Comment by sadath-12** [comment]
> @sadath-12 The other thing here is that we need to keep in mind that these utils functions are being used by the CloudProvider's as well. We should try and open PRs against `Azure/karpenter` and `aws/karpenter-provider-aws` that port over this change after we merge in this cleanup

yes I'll keep that in mind thanks

**Comment by sadath-12** [comment]
Thanks done @jonathan-innis

**Comment by jonathan-innis** [comment]
> Thanks done

**Comment by sadath-12** [comment]
> > Thanks done
> 
> I don't see an updated description in this PR. Did you perhaps forget to press submit here?

yeah sorry about that

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**